suprb/optimizer/rule/ns/ns.py

```diff
@@ -122,13 +99,18 @@ def optimize(self, X: np.ndarray, y: np.ndarray, n_rules: int) -> list[Rule]:
             Return: A list of filtered Rules.
         """
         self._validate_params(n_rules)
-
         self.random_state_ = check_random_state(self.random_state)
+        self.novelty_calculation.archive.set_random_state(self.random_state_)
```

Comment:
Because the `RandomState` is not created at init. If we decide to do `check_random_state` within `Archive` (see above review comment) we can move this into init